dcc-import-gene/src/main/java/org/icgc/dcc/imports/gene/reader/BaseReader.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016 The Ontario Institute for Cancer Research. All rights reserved.                             
+ *                                                                                                               
+ * This program and the accompanying materials are made available under the terms of the GNU Public License v3.0.
+ * You should have received a copy of the GNU General Public License along with                                  
+ * this program. If not, see <http://www.gnu.org/licenses/>.                                                     
+ *                                                                                                               
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY                           
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES                          
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT                           
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,                                
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED                          
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;                               
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER                              
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.imports.gene.reader;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+import java.util.zip.GZIPInputStream;
+
+import lombok.SneakyThrows;
+import lombok.val;
+
+/**
+ * Helper class for reading files and consuming lines.
+ */
+public final class BaseReader {
+
+  /**
+   * Constants
+   */
+  private static final Pattern TSV = Pattern.compile("\t");
+
+  /**
+   * Reads the lines of a file and feeding them to the provided lambda for processing
+   * @param URI - String URI of the file to read
+   * @param lambda - Lambda encapsulating the callers logic for operating on a line
+   */
+  @SneakyThrows
+  public static void read(String URI, Consumer<String[]> lambda) {
+    val bufferedReader = getReader(URI);
+
+    for (String s = bufferedReader.readLine(); null != s; s = bufferedReader.readLine()) {
```

Comment:
I think you really want Guava's `LineProcessor` here:

http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/io/Resources.html#readLines(java.net.URL,%20java.nio.charset.Charset,%20com.google.common.io.LineProcessor)